src/OrchardCore/OrchardCore.Infrastructure/Documents/DocumentManager.cs

```diff
@@ -165,10 +165,12 @@ public async Task UpdateAsync(TDocument document, Func<TDocument, Task> afterUpd
                 {
                     _ = await _distributedCache.GetStringAsync(_options.CacheIdKey);
                 }
-                catch
+                catch(Exception e)
                 {
                     await DocumentStore.CancelAsync();
 
+                    _logger.LogError(e, "Failed to update the '{DocumentName}'", typeof(TDocument).Name);
+                    
                     throw new InvalidOperationException($"Can't update the '{typeof(TDocument).Name}' if not able to access the distributed cache");
                 }
```

Comment:
```suggestion
                catch (Exception ex)
                {
                    await DocumentStore.CancelAsync();

                    throw new InvalidOperationException(
                        $"Can't update the '{typeof(TDocument).Name}' if not able to access the distributed cache",
                        innerException: ex);
                }
```

Then we can do the same in `VolatileDocumentManager.cs`.

Reply:
+1, but log an exception may need too

Reply:
Logging the error would be needed if we do a catch and then don't re-throw, but here the whole exception that we throw will be naturally logged.

Here I needed to catch to do a `CancelAsync()`, and then the idea was to not overfill the log file with the full exception message when you can't connect to redis, because in that case you have the same logged exception from other places, but maybe still useful when the issue is intermitent.

So another suggestion would be to just do the following, where we do a `CancelAsync()` as needed, and where the exception is re-thrown and then will be logged as is.

                catch
                {
                    await DocumentStore.CancelAsync();

                    throw;
                }

Or do the following if still useful to have an higher specific message, knowing that the initial goal of this specific message was to replace the long exception message by a shorter one.

                catch
                {
                    await DocumentStore.CancelAsync();

                    _logger.LogError("Can't update the '{DocumentName}' if not able to access the distributed cache", typeof(TDocument).Name);

                    throw;
                }

Note: In some rare context (not in a given request/bg task), e.g. a kind of background job, an exception may not be naturally logged, in that case this is at an higher level (e.g. a job manager) that the exception needs to be catched and logged, and then re-thrown or not depending on the usage.

Reply:
Hmm, maybe the last suggestion is the best compromise, in the catch block we do the cancel that we need, then we are sure to have a looged error as we do it explicitly, then we just re-throw the original exception.

Reply:
Can we expose the exception object to the logger?

Reply:
But then it will be logged twice, one time here explicitly, another time as we re-throw, like any other exception that we don't catch ourselves but are still logged as is at the end of the request.